californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -715,13 +705,34 @@ private void reject(Message message) {
 	 * @param task the task
 	 */
 	private void runInProtocolStage(final Runnable task) {
-		executor.execute(new Runnable() {
-			public void run() {
-				try {
-					task.run();
-				} catch (Throwable t) {
-					LOGGER.log(Level.SEVERE, "Exception in protocol stage thread: "+t.getMessage(), t);
+		
```

Comment:
I'm not sure, that, if a endpoint is shutdown, the executor should be restarted. If someone depends on using it's own executor, this may also cause troubles. 

So, I would prefer, if you write, how you use californium, that a shutdown of the executor causes issues.

Reply:
Please take a look at this Issue:
https://github.com/eclipse/smarthome/issues/3867

Basically we extend ```CoapClient```, then create an Endpoint using ```public CoapEndpoint(Connector connector, NetworkConfig config)```

and then - issue asynchronous get and put request

Reply:
So, how did you ensure, that the coap endpoint is used proper?
If you shut it down, why do you still use it for get()? 
It looks pretty much for a life-cylce problem on the usage rather the a issue in californium.

Reply:
See

https://github.com/eclipse/smarthome/issues/3867#issuecomment-338983118

Reply:
I agree with @boaks - if the client is shut down, this instance must not be used anymore. Re-initializing it automatically feels pretty wrong.

Reply:
OK - changed it a little bit - the initial tests were with only this change and there were no exceptions